blog/2024-08-19-gsoc-vertx-proxy-updates.mdx

```diff
@@ -0,0 +1,123 @@
+requests. This method enables the injection of custom context content at runtime. For example, to register an proxy 
+as a handler for a http server with custom context, you can using the following code:
+
+```java
+server.requestHandler(request -> {
+    Map<String, Object> customContext = new HashMap<>(); // inject any additional data here
+    proxy.handle(request, customContext);
+});
+```
+
+## GSoC Experience & Learnings
+
+The most interesting (and hard) of this project was the design process. Most of my work involved adding new features 
+to an existing codebase, which required a lot of API design. The challenge was ensuring that the API was easy to 
+understand and use while avoiding any disruptive changes to the original code, which significantly limited the final 
+feasible design options.
+
+For example, there was an issue regarding making the Interceptor support WebSocket. In the original code, WebSocket 
+packets and regular HTTP packets were handled separately, so only the branch processing HTTP packets applied the 
+interceptors. This was reasonable because the handling methods for the two were entirely different. However, the 
+user's request to implement this feature was also reasonable, as the need to modify the WebSocket path exists. 
+Balancing these two requirements took a lot of time during the "finding a feasible solution" phase.
+
+Even after the feature was implemented, I had to consider user experience. If the interceptor were to handle all 
+types of requests by default, users would need to add a condition to all regular interceptors to skip WebSocket 
+traffic, as most request transformations shouldn't apply to them. Therefore, I have to think about a solution where 
+the interceptor doesn't affect WebSocket by default, which also took considerable time.
+
+In fact, the effective time spent writing code was only one day or even half a day, while the design time was one 
+to two times that. So, if I had to summarize the most important lessons:
+
+- Design process is more important than coding; it's best to explicitly allocate time for it during planning.
+- It's better to seek feedback after completing the design immediately rather than jumping straight into coding, as this can easily lead to unexpected outcomes.
+
+And finally, I would like to thank my mentor, Thomas Segismont, for his helpful advice and support throughout the project.
+
+## References
+
+All relevant issues are listed in here. Related PRs are linked in them:
+
+- [issue/67: Cache SPI](https://github.com/eclipse-vertx/vertx-http-proxy/issues/67)
+- [issue/70: HTTP Cache Mechanism](https://github.com/eclipse-vertx/vertx-http-proxy/issues/70)
+- [issue/71: Contextual Data](https://github.com/eclipse-vertx/vertx-http-proxy/issues/71)
+- [issue/48: interceptors for WebSocket](https://github.com/eclipse-vertx/vertx-http-proxy/issues/48)
+- [issue/65: Out-of-the-box interceptors](https://github.com/eclipse-vertx/vertx-http-proxy/issues/65)
```

Comment:
This section will no longer be needed if the text contains link to the issues/PRs